### PR TITLE
Dark mode, flat UI, alignment fixes

### PR DESCRIPTION
**Pull request by pedromvpg**
UI improvements and consistency pass:

- a darker Dark Mode 🦇
- reduce contrast in active button on main nav
- add hover states in main nav
- adjustments on UI to make the design more minimal and flat
  - remove shadows and gradients
  - remove bottom lines from fields
  - standardize colors of divider lines
- alignment issue fix in settings page in national currencies table
- alignment issue in slide toggles
- fix color disparity on support chat (dark mode)

Bisq_UI_Changes

Screen Shot 2020-04-05 at 6 54 58 PM

chat_arrows

### CODE REVIEW

**Comment by wiz** [comment]
Pedro, excellent work on these changes. I know how hard it is to work with Bisq CSS and you did a great job. I just tested your PR and it looks great, but I think removing the lines from text input fields might not be a good idea because now I can't easily tell which text field is focused for text input, and also there are many lines still remaining in other text input fields in other places in the app. Was your intention to only remove them for pull-down combo boxes?

**Comment by wiz** [comment]
Also, here are the CSS errors from the log
```
Apr 06, 2020 4:27:53 PM javafx.scene.CssStyleHelper calculateValue
WARNING: Could not resolve '-bs-rd-nav-deselected' while resolving lookups for '-fx-fill' from rule '*.nav-button *.text' in stylesheet jar:file:/Applications/Bisq.app/Contents/Java/Bisq-1.2.9.jar!/bisq/desktop/bisq.css
Apr 06, 2020 4:27:53 PM javafx.scene.CssStyleHelper calculateValue
WARNING: Could not resolve '-bs-rd-nav-selected' while resolving lookups for '-fx-fill' from rule '*.nav-button:selected *.text' in stylesheet jar:file:/Applications/Bisq.app/Contents/Java/Bisq-1.2.9.jar!/bisq/desktop/bisq.css
Apr 06, 2020 4:27:54 PM javafx.scene.CssStyleHelper calculateValue
WARNING: Could not resolve '-bs-rd-nav-deselected' while resolving lookups for '-fx-fill' from rule '*.nav-button *.text' in stylesheet jar:file:/Applications/Bisq.app/Contents/Java/Bisq-1.2.9.jar!/bisq/desktop/bisq.css
Apr 06, 2020 4:27:54 PM javafx.scene.CssStyleHelper calculateValue
WARNING: Could not resolve '-bs-rd-nav-deselected' while resolving lookups for '-fx-fill' from rule '*.nav-button *.text' in stylesheet jar:file:/Applications/Bisq.app/Contents/Java/Bisq-1.2.9.jar!/bisq/desktop/bisq.css
Apr 06, 2020 4:27:54 PM javafx.scene.CssStyleHelper calculateValue
WARNING: Could not resolve '-bs-rd-nav-selected' while resolving lookups for '-fx-fill' from rule '*.nav-button:selected *.text' in stylesheet jar:file:/Applications/Bisq.app/Contents/Java/Bisq-1.2.9.jar!/bisq/desktop/bisq.css
```

**Comment by wiz** [comment]
Yeah, you can't hide those lines... Here's an example of what is breaking

Expected
Screen Shot 2020-04-06 at 19 55 39

Actual
Screen Shot 2020-04-06 at 19 53 03

the red text is now invisible

**Comment by ghost** [comment]
#4019 is a css issue, if you have time to look into it.

**Comment by pedromvpg** [comment]
@wiz Thanks for the comments and review. I'll resolve the issues on the error log! Should be easy.
Can you tell me how to access that log?

Regarding the fields. Let me see if I can create an alternative that maybe adds a border around the whole field when focused and when invalid. I think it's a more elegant design. Will address it ASAP.

@jmacxx I'll try to resolve this one too! thanks for the heads up!!

OPTION 1 - Outlines
Fields Option 1


OPTION 2 - Less lines, less red
Fields Option 2

**Comment by pedromvpg** [comment]
Added a focus state, an error state, and put back the invalid helper text.

Screen Shot 2020-04-07 at 11 15 12 AM

**Comment by pedromvpg** [comment]
> #4019 is a css issue, if you have time to look into it.

There's something strange overriding the definition of the color on darkmode. Need a little longer to fix this.

**Comment by ghost** [comment]
See also https://github.com/bisq-network/bisq/issues/3957#issuecomment-611846109 when you get the chance.

File: desktop/src/main/java/bisq/desktop/theme-dark.css [hunk 1]
```diff
@@ -266,5 +263,21 @@
...
+}
+
+
+
+
+
+
+
+
+
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/theme-dark.css:282]: Is there a reason for that blank lines?

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 2]
```diff
@@ -65,8 +65,8 @@
 }
 
 .headline-label {
-    -fx-font-weight: bold;
-    -fx-font-size: 1.692em;
+    /*-fx-font-weight: bold;*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: Just delete if if not needed anymore

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 3]
```diff
@@ -154,7 +155,7 @@
     -fx-pref-height: 32;
     -fx-min-height: -fx-pref-height;
     -fx-padding: 0 40 0 40;
-    -fx-effect: dropshadow(gaussian, -bs-text-color-transparent, 2, 0, 0, 0, 1);
+    /*-fx-effect: dropshadow(gaussian, -bs-text-color-transparent, 2, 0, 0, 0, 1);*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: Same here - just delete it if not required anymore

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 4]
```diff
@@ -197,44 +198,58 @@
...
+    -fx-border-radius: 1;
     -fx-border-width: 1;
 }
 
+
 .jfx-combo-box {
     -jfx-focus-color: -bs-color-primary;
     -jfx-unfocus-color: -bs-color-gray-line;
     -fx-background-color: -bs-background-color;
+    /*-jfx-disable-animation: true;*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: Do you want to disable animations only for the combobox?

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 5]
```diff
@@ -197,44 +198,58 @@
...
 }
 
+
+.combo-box-popup > .list-view{
+    -fx-background-color:  -bs-background-color;
+}
+
+
+
+
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: Are those empy lines by intent?

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 6]
```diff
@@ -197,44 +198,58 @@
...
 .jfx-combo-box > .list-cell {
     -fx-text-fill: -bs-text-color;
     -fx-font-family: "IBM Plex Sans Medium";
 }
 
+
+
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: Are those empy lines by intent?

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 7]
```diff
@@ -273,22 +317,39 @@ tree-table-view:focused {
 .jfx-text-field {
     -jfx-focus-color: -bs-color-primary;
     -fx-background-color: -bs-background-color;
-    -fx-background-radius: 3 3 0 0;
+    -fx-background-radius: 4;
     -fx-padding: 0.333333em 0.333333em 0.333333em 0.333333em;
+    /*-jfx-disable-animation: true;*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: Didn't it work doing it like that?

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 8]
```diff
@@ -273,22 +317,39 @@ tree-table-view:focused {
...
+  -fx-text-fill: -bs-rd-error-red;
+  -fx-background-color: derive(-bs-rd-error-field, -5%);
 }
 
+
+
+
+
+
+
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: Are those empy lines by intent?

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 9]
```diff
@@ -312,25 +373,37 @@ tree-table-view:focused {
...
+.jfx-password-field > .input-focused-line {
+    -fx-translate-x: -0.333333em;
+    -fx-background-color: transparent;
+    -jfx-disable-animation: true;
+}
+
+
+
+
+
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: Are those empy lines by intent?

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 10]
```diff
@@ -347,10 +420,15 @@ tree-table-view:focused {
     -fx-font-size: 1em;
 }
 
+
+.field-label {
+  /*-fx-text-fill: red;*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: I guess that was still for testing, based on our last call?

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 11]
```diff
@@ -430,6 +508,9 @@ tree-table-view:focused {
 .jfx-toggle-button:focused:selected {
     -jfx-toggle-color: -bs-color-primary-dark;
     -jfx-size: 8;
+    -jfx-disable-visual-focus: true;
+    /*-jfx-disable-animation: true;*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: Didn't it work as expected?

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 12]
```diff
@@ -590,7 +673,9 @@ tree-table-view:focused {
 }
 
 .content-pane {
-    -fx-background-color: linear-gradient(-bs-content-pane-bg-top 0%, -bs-content-pane-bg-bottom 100%);
+    /*-fx-background-color: linear-gradient(-bs-content-pane-bg-top 0%, -bs-content-pane-bg-bottom 100%);*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: If you don't want it just delete it :-)

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 13]
```diff
@@ -674,14 +760,29 @@ tree-table-view:focused {
 
 .nav-button:selected {
     -fx-background-color: -bs-background-color;
+    -fx-background-color: derive(-bs-color-primary-dark, -10%);
     -fx-border-radius: 4;
-    -fx-effect: dropshadow(gaussian, -bs-text-color-transparent, 4, 0, 0, 0, 2);
+    /*-fx-effect: dropshadow(gaussian, -bs-text-color-transparent, 4, 0, 0, 0, 2);*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: If you don't want it just delete it :-)

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 14]
```diff
@@ -637,7 +722,7 @@ tree-table-view:focused {
 .nav-price-balance {
     -fx-background-color: -bs-color-gray-background;
     -fx-background-radius: 3;
-    -fx-effect: innershadow(gaussian, -bs-text-color-transparent, 3, 0, 0, 1);
+    /*-fx-effect: innershadow(gaussian, -bs-text-color-transparent, 3, 0, 0, 1);*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: If you don't want it just delete it :-)

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 15]
```diff
@@ -1158,12 +1272,12 @@ textfield */
 }
 
 #form-header-text {
-    -fx-font-weight: bold;
-    -fx-font-size: 1.077em;
+    /*-fx-font-weight: bold;*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: If you don't want it just delete it :-)

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 16]
```diff
@@ -1158,12 +1272,12 @@ textfield */
...
+    -fx-font-size: 2.077em;
 }
 
 #form-title {
-    -fx-font-weight: bold;
+    /*-fx-font-weight: bold;*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: If you don't want it just delete it :-)

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 17]
```diff
@@ -1187,9 +1301,11 @@ textfield */
 
 /* Account */
 #content-pane-top {
+    /*
     -fx-background-color: -bs-color-gray-2,
     linear-gradient(-bs-color-gray-2 0%, -bs-color-gray-3 100%),
     linear-gradient(-bs-color-gray-2 0%, -bs-background-gray 100%);
+    */
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: If you don't want it just delete it :-)

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 18]
```diff
@@ -1275,7 +1391,7 @@ textfield */
 }
 
 .combo-box-editor-bold {
-    -fx-font-weight: bold;
+    /*-fx-font-weight: bold;*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: If you don't want it just delete it :-)

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 19]
```diff
@@ -1510,6 +1626,8 @@ textfield */
     -fx-background-color: -bs-background-color;
 }
 
+
+
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css]: Empty lines by intent?

File: desktop/src/main/java/bisq/desktop/theme-dark.css [hunk 20]
```diff
@@ -130,43 +142,27 @@
     -bs-white: white;
     -bs-prompt-text: -bs-color-gray-3;
 
+    /*-bs-decimals: #313133;*/
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/theme-dark.css:145]: Just delete it

**Comment by ripcurlx** [comment]
Did a quick smoke test on the UI changes and I do like them I only found following issues/have following concerns:

**Light mode**
I think here without the even odd colors you want to add the separation lines as well?
Bildschirmfoto 2020-04-28 um 16 03 07
The background color of the button in disabled style is hardly recognizable.
Bildschirmfoto 2020-04-28 um 16 05 51
The layout of the fee toggle buttons is broken.
Bildschirmfoto 2020-04-28 um 16 07 27
I guess you want the error background on the whole field.
Bildschirmfoto 2020-04-28 um 16 08 31
Here the layout is also broken a little bit.
Bildschirmfoto 2020-04-28 um 16 09 44

**Dark mode**
When taking an offer there is this thick dark grey line and the toggle buttons are out of place here as well
Bildschirmfoto 2020-04-28 um 16 13 26

**Comment by ripcurlx** [comment]
Maybe lets have a chat where I can point you in the right direction for the last few issues and I'm happy to try out this new style adaption.

**Comment by wiz** [comment]
hey @pedromvpg wanna talk about this

**Comment by cbeams** [comment]
>  [Create MempoolFeeRateProvider 2.java](/bisq-network/bisq/pull/4121/commits/1e86ff65dfb17df9f51b2bd007d39ccf04462136) 

Hm... looks like this was an accidental addition; correct, @pedromvpg?

**Comment by pedromvpg** [comment]
I screwed up this pull request. Most likely will have to start over and break it down into smaller scopes.

**Comment by wiz** [comment]
I got you bro. See PR #4334

**Pull request closed without merging**